Source/Atomic/Engine/Engine.cpp

```diff
@@ -558,6 +566,23 @@ void Engine::SetNextTimeStep(float seconds)
     timeStep_ = Max(seconds, 0.0f);
 }
 
+void Engine::SetPaused(bool paused)
+{
+    paused_ = paused;
+
+    using namespace UpdatesPaused;
+
+    // Updates paused event
+    VariantMap& eventData = GetEventDataMap();

```

Comment:
Nice reuse of event data map, one note, this is generally only needed for methods that have a high call count